crypto/hash_test.go

```diff
@@ -1,29 +1,38 @@
 package crypto
 
 import (
+	"bytes"
+	"encoding/hex"
 	"testing"
 )
 
 func TestBlake(t *testing.T) {
-	r1 := Blake256("abcd")
-	if r1 != "9606e52f00c679e548b5155af5026f5af4130d7a15c990a791fff8d652c464f5" {
+	input, _ := hex.DecodeString("abcd")
+	r1 := Blake256(input)
+	a1 := []byte("9606e52f00c679e548b5155af5026f5af4130d7a15c990a791fff8d652c464f5")
+	if 0 != bytes.Compare(r1, a1) {
 		t.Fatal("Blake256 function output error")
 	}
 
 	key1 := []byte{1, 2, 3, 4}
-	r2 := Blake128WithKey("abcd", key1)
-	if r2 != "cc98decf76abbbdf3a2027552bec09c8" {
+	r2 := Blake128WithKey(input, key1)
+	a2 := []byte("cc98decf76abbbdf3a2027552bec09c8")
+	if 0 != bytes.Compare(r2, a2) {
 		t.Fatal("Blake128WithKey function output error")
 	}
 
+	input2, _ := hex.DecodeString("aabbccddeeff0011223344")
 	key2 := []byte{33, 65, 97, 34, 35, 36}
-	r3 := Blake160WithKey("aabbccddeeff0011223344", key2)
-	if r3 != "68e2afd80441be243c755c2184eb4c9ce9f9f17c" {
+	r3 := Blake160WithKey(input2, key2)
+	a3 := []byte("68e2afd80441be243c755c2184eb4c9ce9f9f17c")
+	if 0 != bytes.Compare(r3, a3) {
 		t.Fatal("Blake160WithKey function output error")
 	}
 
-	r4 := Ripemd160("abcdef12345678")
-	if r4 != "6b1f9162346b8962edde6e28ee7599541def053e" {
+	input3, _ := hex.DecodeString("abcdef12345678")
+	r4 := Ripemd160(input3)
+	a4 := []byte("6b1f9162346b8962edde6e28ee7599541def053e")
```

Comment:
Casting a string typed value to byte array type uses UTF8 encoding. Please use hex.DecodeString.